src/dissect/dissect.c

```diff
@@ -842,6 +862,31 @@ static int action_copy(DissectedImage *m, LoopDevice *d) {
                 (void) copy_times(source_fd, target_fd, 0);
 
                 /* When this is a regular file we don't copy ownership! */
+
+        } else {
+                _cleanup_strv_free_ char **list_dir = NULL;
+                size_t offset = strlen(mounted_dir);
+
+                int list_print_item(
+                                RecurseDirEvent event,
+                                const char *path,
+                                int dir_fd,
+                                int inode_fd,
+                                const struct dirent *de,
+                                const struct statx *sx,
+                                void *userdata) {
+
+                        assert_se(path);
+
+                        if (IN_SET(event, RECURSE_DIR_ENTER, RECURSE_DIR_ENTRY))
+                                printf("%s\n", path + offset);
```

Comment:
hmm, I think it would make sense to suffix directories here with a "/". I know that "tar l" doesn't do that, but I still would do that, it's a really useful hint to distinguish directories from other types of inodes, without "dirtying" the output in a way that the listed paths weren#t valid anymore.


i.e. i want this output:

```
foo/
foo/bar
foo/quux/
```

for an image where `foo` and `foo/quux` are dirs, and `/foo/bar` is a regular file (or other type of inode)